pkg/socket/client.go

```diff
@@ -0,0 +1,62 @@
+package socket
+
+import (
+	"bufio"
+	"crypto/tls"
+	"net"
+	"time"
+)
+
+func NewSocket(config Config) *socket {
+	return &socket{
+		Config: config,
+		conn:   nil,
+	}
+}
```

Comment:
I think we need either return public method or interface. This won't do if we add documentation - we can documents what "socket" is.

Reply:
Ok, so I've made Socket in order to provide documentation to the users.